app_dart/lib/src/service/branch_service.dart

```diff
@@ -114,8 +115,17 @@ class BranchService {
       retryIf: (Exception e) => e is gh.GitHubError,
     );
     log.info('${Config.engineSlug} branch commits: $githubCommits');
+    final DateTime? branchTime = githubCommits
+        .firstWhereOrNull((gh.RepositoryCommit commit) => commit.commit?.committer?.date != null)
+        ?.commit
+        ?.committer
+        ?.date;
+    if (branchTime == null) {
+      throw BadRequestException('Found no GitHub commits on $branch with commit time');
+    }
     for (GerritCommit recipeCommit in recipeCommits) {
-      if (recipeCommit.author!.date!.isBefore(githubCommits.first.commit!.committer!.date!)) {
+      final DateTime? recipeTime = recipeCommit.author?.date;
+      if (recipeTime != null && recipeTime.isBefore(branchTime)) {
```

Comment:
Umm quick check, are `recipeCommits` and `githubCommits` already sorted by newest time to oldest time?